From 71fb43adb07dd6807e776a107ccf6b6a85821883 Mon Sep 17 00:00:00 2001
From: FedericoYundt <FedericoYundt@users.noreply.github.com>
Date: Wed, 18 Apr 2018 08:40:59 +0100
Subject: Allow cupsd daemon to exit on idle

When using socket activation cups daemon can safely exit on idle and be automatically started when it's needed. Using Restart=always prevent this behavior and causes cups daemon constantly restarting itself instead of exiting. Users reported[1] that Restart=on-failure was enough to fix issues[2] so Restart=always may be unnecessary and harmful.

[1] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=861470
[2] https://github.com/apple/cups/issues/5263

Bug-Upstream: https://github.com/apple/cups/pull/5297
Bug-Debian: https://bugs.debian.org/898745
---
 scheduler/org.cups.cupsd.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scheduler/org.cups.cupsd.service.in b/scheduler/org.cups.cupsd.service.in
index 832169586..3c273612a 100644
--- a/scheduler/org.cups.cupsd.service.in
+++ b/scheduler/org.cups.cupsd.service.in
@@ -5,7 +5,7 @@ Documentation=man:cupsd(8)
 [Service]
 ExecStart=@sbindir@/cupsd -l
 Type=simple
-Restart=always
+Restart=on-failure
 
 [Install]
 Also=cups.socket cups.path
